src/junit.rs

```diff
@@ -126,26 +131,31 @@ pub fn parse_junit_xml(file_bytes: &[u8]) -> PyResult<Vec<Testrun>> {
                     curr_testsuite = get_attribute(&e, "name")?
                         .ok_or(ParserError::new_err("Error getting name".to_string()))?;
                 }
+                b"testsuites" => {
+                    testsuites_name = get_attribute(&e, "name")?;
+                }
                 _ => {}
             },
-            Ok(Event::End(e)) => match e.name().as_ref() {
+            Event::End(e) => match e.name().as_ref() {
                 b"testcase" => {
-                    list_of_test_runs.push(
-                        saved_testrun
-                            .ok_or(ParserError::new_err("Error accessing saved testrun"))?,
-                    );
+                    let testrun = saved_testrun.ok_or(ParserError::new_err(
+                        "Met testcase closing tag without first meeting testcase opening tag"
+                            .to_string(),
+                    ))?;
```

Comment:
I think `ParserError` is a "class" extended from python, so potentially quite heavy.
The `ok_or` would create that struct/class ahead of time, even when there is no error.
You can use `ok_or_else` to pass a closure, to construct it only on-demand.

The same also applies to the two cases in `populate` as well.

Also, the `new_err` constructor works with a `&str` as well, so you can avoid the `to_string`.